src/models/ConstrainedMetaModel.ts

```diff
@@ -48,20 +48,23 @@ export class ConstrainedTupleModel extends ConstrainedMetaModel {
   }
 
   getNearestDependencies(): ConstrainedMetaModel[] {
-    let dependencyModels = Object.values(this.tuple).filter(
-      (tupleModel) => {
-        return tupleModel.value instanceof ConstrainedReferenceModel;
+    let dependencyModels: ConstrainedMetaModel[] = [];
+    for (const tupleModel of Object.values(this.tuple)) {
+      if (tupleModel.value instanceof ConstrainedReferenceModel) {
+        dependencyModels.push(tupleModel.value);
+      } else {
+        //Lets check the non-reference model for dependencies
+        dependencyModels = [...dependencyModels, ...tupleModel.value.getNearestDependencies()];
       }
-    ).map((tupleModel) => {
-      return tupleModel.value = tupleModel.value as ConstrainedReferenceModel;
+    }
+    
+    dependencyModels = dependencyModels.map((model) => {
+      return model as ConstrainedReferenceModel;
     });
```

Comment:
What?! 😄

Reply:
You should cast returned type to the `ConstrainedReferenceModel[]` if you have errors.

Reply:
Brainfart removed the unused code 😄